### PR TITLE
Add registerFilter method

### PR DESCRIPTION
**Pull request by didoo**
*Issue #, if available:* #207 #199 

*Description of changes:*
- introduced a `registerFilter` method for the StyleDictionary module
- updated the `filterProperties.js` to receive the `filter` parameter only as a function
- moved the logic to handle the case when the `filter` parameter is an object in the `transformConfig` method, and added the handling of the case where the filter is a string (supposedly a filter registered via `registerFilter`, otherwise throw an error)
- updated the documentation accordingly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by didoo** [comment]
@dbanksdesign @chazzmoney before continuing this PR (updating tests and documentation) I need some feedback from you. Have a look at the code, it works but maybe (probably) can be better.

Things I am not 100% sure:

1) I have moved the logic where we convert the filter from object/function to function (and added the case for the string (registered function) from the `filterProperties` method to the `transformConfig` one. The reason for that is that in filterProperties I don't have access to the StyleDictionary object (where the filters are registered). I have tried to pass it down, or to use `this` but without success. Do you think there would be a way to do it? Where is the "correct" location for this logic?
https://github.com/amzn/style-dictionary/blob/b7d1672b163e5941e860bd49ba7e3b8ce55031a0/lib/transform/config.js#L55-L70

2) I have left a comment, about the initialization of the `filter` object as a property of StyleDictionary: 
https://github.com/amzn/style-dictionary/blob/b7d1672b163e5941e860bd49ba7e3b8ce55031a0/lib/register/filter.js#L38
I get why we need to initialise it (templates and formats are always present in SD, filter not necessarily) but I am not sure if there is a place where is better to initialise it. Any better ideas?

**Comment by dbanksdesign** [comment]
For question 1, you are correct in the placement. `transform/config.js` is meant to 'hydrate' the config object so it has functions rather than string keys for transforms/formats/actions/etc. Adding filters, this would be the proper place to do it. 

For question 2, the initialization happens in the main index.js file:
https://github.com/amzn/style-dictionary/blob/master/index.js#L36
The file that it requires returns an object. So we could just add:
```javascript
filter: {}
```
in there and call it a day. Or do we want to include built-in filters? Maybe not if we are trying to distribute/open up the transforms/formats/actions. Sorry for stream of conscious here.

**Comment by didoo** [comment]
> question 1

Cool, I am happy then! :)

> question 2

Ahh, now I get it! Now that you have explained it, makes total sense. OK, so we have these options:

1.  we leave the "initialisation" in the `lib/register/filter.js` file, but is an antipattern
2. we add a line in `index.js` like you suggested (with a comment why is done like this)
3. we create an "empty" module `lib/common/filters` that returns an empty object, so is consistent with the others (we need to add a comment in the module anyway, and possibly also in the `index.js` to signal that that is a special module

Probably 2 is the simplest and clearest one. What do you think?

**Comment by dbanksdesign** [comment]
I like 2 or 3. We can just go with 2, and then if we do add built-in filters in the future we can update it when that happens.

File: __tests__/filterProperties.test.js [hunk 1]
```diff
@@ -112,7 +112,7 @@ describe('filterProperties', () => {
     expect(filteredDictionary.properties).not.toHaveProperty('color');
   });
 
-  it('should work with a filter object', () => {
+  xit('should work with a filter object', () => {
```
Comment (dbanksdesign) [inline __tests__/filterProperties.test.js]: typo?

Comment (didoo) [reply __tests__/filterProperties.test.js]: It was a skipped test (this is how you tell Jest to skip a test). I had forgot to fix it 😬
Now I have updated it correctly for the way the function works now.

Comment (dbanksdesign) [reply __tests__/filterProperties.test.js]: Ohhh... TIL

**Comment by didoo** [comment]
@dbanksdesign wait before merging, I still have to update the documentation :)

**Comment by dbanksdesign** [comment]
Good call, I might have merged today by accident

**Comment by didoo** [comment]
@dbanksdesign I have updated the documentation files. Do you think is enough?

**Comment by chazzmoney** [comment]
LGTM

**Pull request merged**